lib/types.h

```diff
@@ -105,9 +79,9 @@ template <class Archive> void serialize(Archive& archive, Color& c) {
  */
 struct Box {
     float surface_area() const {
-        auto e0 = max[Axis::X] - min[Axis::X];
-        auto e1 = max[Axis::Y] - min[Axis::Y];
-        auto e2 = max[Axis::Z] - min[Axis::Z];
+        auto e0 = max[0] - min[0];
```

Comment:
How about a const to name the Axis?

I just saw the Axis type. I guess `max[static_cast<size_t>(Axis.X)]` is a little overhead ;).